test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SpanExpectation.cs

```diff
@@ -10,19 +10,20 @@ namespace Datadog.Trace.ClrProfiler.IntegrationTests
     /// </summary>
     public class SpanExpectation
     {
-        public SpanExpectation(string serviceName, string operationName, string type)
+        public SpanExpectation(string serviceName, string operationName, string resourceName, string type)
         {
             ServiceName = serviceName;
             OperationName = operationName;
+            ResourceName = resourceName;
             Type = type;
 
             // Expectations for all spans regardless of type should go here
             RegisterDelegateExpectation(ExpectBasicSpanDataExists);
 
             RegisterCustomExpectation(nameof(OperationName), actual: s => s.Name, expected: OperationName);
             RegisterCustomExpectation(nameof(ServiceName), actual: s => s.Service, expected: ServiceName);
+            RegisterCustomExpectation(nameof(ResourceName), actual: s => s.Resource.ToLowerInvariant().TrimEnd(), expected: ResourceName.ToLowerInvariant());
```

Comment:
Previously, the expected`ResourceName` was set to `null` because the property was not set when the constructor was called, and this caused us to skip that validation rule. This change (and all of the plumbing needed to support the new constructor) allows us to set the expected `ResourceName` to a non-null value.